### ISSUE TITLE
Not able to use nested hierarchies in dynamic inventory

### ISSUE THREAD

**Issue by przemeq84** [ansible/ansible#9262]
##### Issue Type:

Bug Report
##### Ansible Version:

1.7.2
##### Environment:

Control machine: Mac OS X 10.9.5
##### Summary:

It seems that, when parsing an output of dynamic inventory script, groups that contain only other groups (without any vars or hosts) are treated as hosts and added to the host list. This causes an ansible playbook to fail because it tries to connect to host by a group name.

The same functionality works just fine with static inventory files.
##### Steps To Reproduce:

Use the following inventory scripts:

Dynamic inventory (file: _test.py_):

```
test.py
#!/usr/bin/env python                                                                                                                                                                

print  """{ "child": ["host1.example.com"],                                                                                                                                          
            "parent": {                                                                                                                                                              
              "children": [ "child" ]                                                                                                                                                
            }                                                                                                                                                                        
          }"""  
```

Static inventory (file: _test_): 

```
[child]                                                                                                                                                                              
host1.example.com                                                                                                                                                                    

[parent:children]                                                                                                                                                                    
child     
```
##### Expected Results:

Running static inventory returns only the actual host:

```
$ ansible -i test -a "" all --list-hosts
host1.example.com
```
##### Actual Results:

Running dynamic inventory script returns the empty parent group as well:

```
$ ansible -i test.py -a "" all --list-hosts
host1.example.com
parent
```
##### Comments:

I believe this functionality is caused by the https://github.com/ansible/ansible/commit/b15cc3e2b4bd86e46be8d3a95cd7f3c27a98ae40 commit. It seems that this change is supposed to make it possible to add hosts not included in any group in the dynamic inventory script result.

I can prepare a fix for this, but I'm not sure if it's better to revert this change, as I don't see anything about this functionality in the ansible docs, or to add another _if_ statement to make sure that groups won't be added to the hosts list (and leaving the previous if for compatibility reasons).

**Comment by srgvg** [comment]
Same issue as and proposed fixed #8936

**Comment by mpdehaan** [comment]
Thanks, I'm going to close this one as a dup as we have the ticket over there.
